### PR TITLE
OSDOCS#12121:HCP KubeVirt Nvidia GPU support

### PR DESCRIPTION
**Pull request by xenolinux**
Version(s): 4.17+

Issue: https://issues.redhat.com/browse/OSDOCS-12121

Link to docs preview: https://83599--ocpdocs-pr.netlify.app/openshift-enterprise/latest/hosted_control_planes/hcp-manage/hcp-manage-virt.html#hcp-virt-attach-nvidia-gpus_hcp-manage-virt 
https://83599--ocpdocs-pr.netlify.app/openshift-enterprise/latest/hosted_control_planes/hcp-manage/hcp-manage-virt.html#hcp-virt-attach-nvidia-gpus-np-api_hcp-manage-virt

QE review:
- [x] QE has approved this change.

SME review:
- [x] SME has approved this change.

Additional information:

### CODE REVIEW

**Review by davidvossel — COMMENTED** [review]
/lgtm

**Review by davidvossel — COMMENTED** [review]
/lgtm

File: modules/hcp-virt-attach-nvidia-gpus-np-api.adoc [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        type: Persistent
+    type: KubeVirt
+  replicas: <worker_node_count> <8>
+----
+<1> Specify the name of your hosted cluster, for instance, `example`.
+<2> Specify the name of the hosted cluster namespace, for example, `clusters`.
+<3> Specify a value for CPU, for example, `2`.
+<4> Specify a value for memory, for example, `16Gi`.
+<5> The `hostDevices` field defines a list of different types of GPU devices that you can attach to node pools.
+<6> Specify the number of GPU devices you want to attach to each virtual machine (VM) in node pools, for example, if you attach 2 GPU devices to 3 node pool replicas, all 3 VMs in the node pool are attached 2 GPU devices. The default count is `1`.
```
Comment (lahinson) [inline modules/hcp-virt-attach-nvidia-gpus-np-api.adoc:57]:
```suggestion
<6> Specify the number of GPU devices you want to attach to each virtual machine (VM) in node pools. For example, if you attach 2 GPU devices to 3 node pool replicas, all 3 VMs in the node pool are attached to the 2 GPU devices. The default count is `1`.
```

File: modules/hcp-virt-attach-nvidia-gpus.adoc [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+= Attaching NVIDIA GPU devices by using the hcp CLI
+
+You can attach one or more NVIDIA graphics processing unit (GPU) devices to node pools by using the `hcp` command-line interface (CLI) in a hosted cluster on {VirtProductName}.
+
+:FeatureName: Attaching NVIDIA GPU devices to node pools
+include::snippets/technology-preview.adoc[]
+
+.Prerequisites
+
+* You have exposed the NVIDIA GPU device as a resource on the node that the GPU device resides on. For more information, see link:https://docs.nvidia.com/datacenter/cloud-native/openshift/latest/openshift-virtualization.html[NVIDIA GPU Operator with {VirtProductName}].
```
Comment (lahinson) [inline modules/hcp-virt-attach-nvidia-gpus.adoc:16]:
```suggestion
* You have exposed the NVIDIA GPU device as a resource on the node where the GPU device resides. For more information, see link:https://docs.nvidia.com/datacenter/cloud-native/openshift/latest/openshift-virtualization.html[NVIDIA GPU Operator with {VirtProductName}].
```

File: modules/hcp-virt-attach-nvidia-gpus.adoc [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  --memory <memory> \// <4>
+  --cores <cpu> \// <5>
+  --host-device-name="<gpu_device_name>,count:<value>" <6>
+----
+<1> Specify the name of your hosted cluster, for instance, `example`.
+<2> Specify the worker count, for example, `3`.
+<3> Specify the path to your pull secret, for example, `/user/name/pullsecret`.
+<4> Specify a value for memory, for example, `16Gi`.
+<5> Specify a value for CPU, for example, `2`.
+<6> Specify the GPU device name and the count, for example, `--host-device-name="nvidia-a100,count:2"`. The `--host-device-name` argument takes the name of the GPU device from the infrastructure node and an optional count that represents the number of GPU devices you want to attach to each virtual machine (VM) in node pools. The default count is `1`. For example, if you attach 2 GPU devices to 3 node pool replicas, all 3 VMs in the node pool are attached 2 GPU devices.
```
Comment (lahinson) [inline modules/hcp-virt-attach-nvidia-gpus.adoc:39]:
```suggestion
<6> Specify the GPU device name and the count, for example, `--host-device-name="nvidia-a100,count:2"`. The `--host-device-name` argument takes the name of the GPU device from the infrastructure node and an optional count that represents the number of GPU devices you want to attach to each virtual machine (VM) in node pools. The default count is `1`. For example, if you attach 2 GPU devices to 3 node pool replicas, all 3 VMs in the node pool are attached to the 2 GPU devices.
```

File: hosted_control_planes/hcp-manage/hcp-manage-virt.adoc [hunk 4]
```diff
@@ -50,3 +50,5 @@ include::modules/hcp-virt-image-caching.adoc[leveloffset=+2]
 * xref:../../virt/virtual_machines/creating_vms_custom/virt-creating-vms-by-cloning-pvcs.adoc#smart-cloning_virt-creating-vms-by-cloning-pvcs[Cloning a data volume using smart-cloning]
 
 include::modules/hcp-virt-etcd-storage.adoc[leveloffset=+2]
+include::modules/hcp-virt-attach-nvidia-gpus.adoc[leveloffset=+2]
```
Comment (lahinson) [inline hosted_control_planes/hcp-manage/hcp-manage-virt.adoc:53]: Should these 2 new topics be set at level 2 or level 1? I'm not sure whether they should be nested under the "Configuring storage..." section.

**Review by lahinson — COMMENTED** [review]
Nice work on this! Just a couple of suggestions and one question for your consideration.

File: hosted_control_planes/hcp-manage/hcp-manage-virt.adoc (hunk 4 above)
Comment (xenolinux) [reply hosted_control_planes/hcp-manage/hcp-manage-virt.adoc]: Right. I think I can shift them to  the level 1.

**Pull request merged**